### PR TITLE
Initial security support in tenant manager

### PR DESCRIPTION
**Pull request by carlesarnal**
This pr adds support for security in the tenant manager, adds the implementation of the client credentials flow to the client and also some tests that tests both the API and the client.

### CODE REVIEW

**Comment by carlesarnal** [comment]
The native images build will probably fail due to the HTTP permissions. We will probably need to port the mechanism that I've implemented for the registry to the tenant-manager.

**Comment by carlesarnal** [comment]
@EricWittmann @famartinrh this is ready for review

File: multitenancy/tenant-manager-api/src/test/java/io/apicurio/multitenant/TenantManagerClientAuthTest.java [hunk 1]
```diff
@@ -0,0 +1,65 @@
...
+import org.junit.jupiter.api.Tag;
+import org.junit.jupiter.api.Test;
+import org.keycloak.authorization.client.util.HttpResponseException;
+
+import javax.enterprise.inject.Typed;
+
+
+@QuarkusTest
+@TestProfile(AuthTestProfileWithoutRoles.class)
+@Tag(ApicurioTestTags.DOCKER)
```
Comment (famartinrh) [inline multitenancy/tenant-manager-api/src/test/java/io/apicurio/multitenant/TenantManagerClientAuthTest.java:37]: if you are adding this tag you need to configure `maven-surefire-plugin` in this projects pom.xml file the same way it's configured in `app` module, basically add this:
```
<build>
  <plugins>
    <plugin>
                <groupId>org.apache.maven.plugins</groupId>
                <artifactId>maven-surefire-plugin</artifactId>
                <configuration>
                    <groups>${groups}</groups>
                    <systemProperties>
                        <java.util.logging.manager>org.jboss.logmanager.LogManager</java.util.logging.manager>
                    </systemProperties>
                </configuration>
            </plugin>
  </plugins>
</build>
```
```
<profiles>
        <profile>
            <id>no-docker</id>
            <properties>
                <groups>!docker</groups>
            </properties>
        </profile>
</profiles>
```

**Review by EricWittmann — CHANGES_REQUESTED** [review]
LGTM just need to make Fabian's suggested change I think.

**Pull request merged**